platform/ios/src/MGLMapView.mm

```diff
@@ -3044,10 +3044,9 @@ - (void)removeStyleClass:(NSString *)styleClass
                 continue;
             }
             MGLAnnotationContext annotationContext = _annotationContextsByAnnotationTag.at(annotationTag);
-            if (annotationContext.annotation)
-            {
-                [annotations addObject:annotationContext.annotation];
-            }
+            NSAssert(annotationContext.annotation, @"Missing annotation for tag %u.", annotationTag);
+            [annotations addObject:annotationContext.annotation];
```

Comment:
https://github.com/mapbox/mapbox-gl-native/issues/8163#issuecomment-290246849 mentions supplementing the fix with an assertion. This change fully removes the guard added in https://github.com/mapbox/mapbox-gl-native/pull/8513. We can keep the assertion although I'm now confident we will hit it (or other similar assertions) from time to time when developing because of what I described in https://github.com/mapbox/mapbox-gl-native/issues/8163#issuecomment-291695262